client/onboarding/add-business-info-task/index.js

```diff
@@ -0,0 +1,171 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { useState, useContext } from 'react';
+import { __ } from '@wordpress/i18n';
+import { SelectControl, Card, CardBody } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import WizardTaskItem from 'wcpay/additional-methods-setup/wizard/task-item';
+import WizardTaskContext from 'wcpay/additional-methods-setup/wizard/task/context';
+import OnboardingSelectControl from 'wcpay/components/onboarding-select-control';
+import { useBusinessTypes } from 'wcpay/data/onboarding';
+import strings from '../strings';
+
+const AddBusinessInfo = () => {
+	const { setCompleted } = useContext( WizardTaskContext );
+	const { businessTypes, isLoading } = useBusinessTypes();
+	const {
+		connect: { availableCountries, country },
+	} = wcpaySettings;
+
+	const countries = Object.keys( availableCountries ).map(
+		( countryCode ) => {
+			return {
+				label: availableCountries[ countryCode ],
+				value: countryCode,
+			};
+		}
+	);
+
+	const [ businessCountry, setBusinessCountry ] = useState( country );
+	const [ businessType, setBusinessType ] = useState();
+	const [ businessStructure, setBusinessStructure ] = useState();
+	const [ displayStructures, setDisplayStructures ] = useState( false );
+
+	const getBusinessTypesForCountry = ( countryCode ) => {
+		if ( isLoading || ! businessTypes.hasOwnProperty( 'data' ) ) {
+			return false;
+		}
```

Comment:
Dealing with objects can get a bit tricky. We could probably simplify this by storing an array of countries in `getBusinessTypes` resolver. We're declaring a type of `Countries[]` but it's an object nested in `data`.

With an array of countries we can get rid of the checks of `data`, and store the selected one directly, not the key to access it later.

Reply:
Thanks, I like this idea! I want to clarify though, does this mean we'd only pass the currently selected country through to our component? How would we handle when the country is changed via the dropdown?

I'm assuming this is dealt with by the resolver somehow, but the picture is a bit unclear in my mind 😅

Reply:
Sorry, I wasn't clear enough! 🙂 

My idea is to make the resolver store an array of countries. Currently, it stores an object like `{data: {AE: {...}, ...}}` and I'm thinking on `['AE': {...}, ...]`. You'll probably need to update the new server PR to return the new format. And you'll receive the array while using the selector, like you already defined with types.

Then you can iterate on that array to create the select options, and when you select a country rather than storing the select country key, you can store the full country object. And later access it to look for the types and structures.

Let me know if it's clearer now, I can make a quick code example if you need it 🙂